cypress/e2e/apiTeste1.cy.js

```diff
@@ -0,0 +1,109 @@
+// tetse exercicios iniciais !!//
+describe('API de Clientes', { defaultCommandTimeout: 10000 }, () => {
+    const API_URL = 'http://localhost:3001'
+    const UI_URL = 'http://localhost:3000'
+  
+    const clickCookieConsent = () => {
+      cy.wait(2000) // Espera 2 segundos para garantir que a página carregou completamente
+      cy.get('body').then($body => {
+        if ($body.find('.CookieConsent_buttons__94qJE > :nth-child(1) > button').length > 0) {
```

Comment:
# Evite lógicas complexas em testes e comandos customizados

É comum alunos/as criarem testes com lógicas complexas, tais como loops e condicionais, tornando o entendimento do que exatamente cada teste verifica bastante difícil.

## Por que tal prática fere um bom design de testes?

Lógicas complexas, como loops e condicionais aumentam a complexidade ciclomática do código, o que dificulta o entendimento completo do que cada pedaço de código faz e fere a regra de que testes devem ser lineares, ou seja, devem possuir pré-condições, ações e resultados esperados.

## Sugestão com um melhor design

Evite lógicas complexas como loops e condicionais ao máximo em testes automatizados.

Exemplo:

```js
beforeEach(() => {
  Cypress.on('window:before:load', window => {
    window.document.cookie = 'cookieConsent=accepted'
  })

  cy.visit('/')
})
```

Se você acha que precisa de um condicional, quem sabe o que você precise seja de um caso de teste (bloco `it`) para cada situação possível de tal condicional.

Reply:
Perceba que no exemplo acima estou definindo os _cookies_ antes mesmo de visitar a aplicação, evitando a necessidade no clique do botão _Accept_.